salt/crypt.py

```diff
@@ -1239,7 +1239,7 @@ def loads(self, data):
         # simple integrity check to verify that we got meaningful data
         if not data.startswith(self.PICKLE_PAD):
             return {}
-        load = self.serial.loads(data[len(self.PICKLE_PAD):])
-        if six.PY3:
+        load = self.serial.loads(data[len(self.PICKLE_PAD):], raw=raw)
+        if six.PY3 and not raw:
             load = salt.transport.frame.decode_embedded_strs(load)
```

Comment:
You changed `self.serial.loads` to call `decode_embedded_strs`. So this logic would double-decode which is inefficient.